api/v1alpha1/environment_conversion.go

```diff
@@ -0,0 +1,123 @@
+		Type:               v1beta1.EnvironmentType(src.Spec.Type),
+		DisplayName:        src.Spec.DisplayName,
+		DeploymentStrategy: v1beta1.DeploymentStrategyType(src.Spec.DeploymentStrategy),
+		ParentEnvironment:  src.Spec.ParentEnvironment,
+		Tags:               src.Spec.Tags,
+	}
+
+	// if v1alpha1 version has src.Spec.Configuration.Env field then copy it to v1beta1
+	if src.Spec.Configuration.Env != nil {
+		for _, env := range src.Spec.Configuration.Env {
+			dst.Spec.Configuration.Env = append(dst.Spec.Configuration.Env, v1beta1.EnvVarPair(env))
+		}
+	}
+
+	// if v1alpha1 version has Spec.Configuration.Target field then copy it to v1beta1
+	if src.Spec.Configuration.Target.DeploymentTargetClaim.ClaimName != "" {
+		dst.Spec.Configuration.Target = v1beta1.EnvironmentTarget{
+			DeploymentTargetClaim: v1beta1.DeploymentTargetClaimConfig{
+				ClaimName: src.Spec.Configuration.Target.DeploymentTargetClaim.ClaimName,
+			},
+		}
+	}
+
+	// if v1alpha1 has Spec.UnstableConfigurationFields field then copy it to v1beta1
+	if src.Spec.UnstableConfigurationFields != nil {
+		dst.Spec.Target = &v1beta1.TargetConfiguration{
+			ClusterType: v1beta1.ConfigurationClusterType(string(src.Spec.UnstableConfigurationFields.ClusterType)),
+		}
+
+		dst.Spec.Target.KubernetesClusterCredentials = v1beta1.KubernetesClusterCredentials{
+			TargetNamespace:            src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.TargetNamespace,
+			APIURL:                     src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.APIURL,
+			IngressDomain:              src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.IngressDomain,
+			ClusterCredentialsSecret:   src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.ClusterCredentialsSecret,
+			AllowInsecureSkipTLSVerify: src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.AllowInsecureSkipTLSVerify,
+			Namespaces:                 src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.Namespaces,
+			ClusterResources:           src.Spec.UnstableConfigurationFields.KubernetesClusterCredentials.ClusterResources,
+		}
+	}
+
+	return nil
+}
+
+// ConvertFrom converts from the Hub version (vbeta1) to this version.
+func (dst *Environment) ConvertFrom(srcRaw conversion.Hub) error {
+
+	src := srcRaw.(*v1beta1.Environment)
+
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec = EnvironmentSpec{
+		Type:               EnvironmentType(src.Spec.Type),
+		DisplayName:        src.Spec.DisplayName,
+		DeploymentStrategy: DeploymentStrategyType(src.Spec.DeploymentStrategy),
+		ParentEnvironment:  src.Spec.ParentEnvironment,
+		Tags:               src.Spec.Tags,
+	}
+
+	if src.Spec.Configuration.Env != nil {
+		for _, env := range src.Spec.Configuration.Env {
+			dst.Spec.Configuration.Env = append(dst.Spec.Configuration.Env, EnvVarPair(env))
+		}
+	}
+
+	if src.Spec.Configuration.Target.DeploymentTargetClaim.ClaimName != "" {
```

Comment:
I think this should be:
```go
	if src.Spec.Configuration.Env != nil {

		dst.Spec.Configuration.Env = []EnvVarPair{}

		for _, env := range src.Spec.Configuration.Env {
			dst.Spec.Configuration.Env = append(dst.Spec.Configuration.Env, EnvVarPair(env))
		}
	}
```
Otherwise we are converting empty slice to nil slice, which is semantically different.